src/index.ts

```diff
@@ -1,100 +1,79 @@
 import { Elysia, t } from "elysia"
-
-let data = [{
-  id: 1,
-  title: 'First Item',
-  description: "Description for the first item",
-  isComplete: false
-},
-{
-  id: 2,
-  title: 'Second Item',
-  description: "Description for the second item",
-  isComplete: false
-},
-{
-  id: 3,
-  title: 'Third Item',
-  description: "Description for the third item",
-  isComplete: false
-}]
+import {cors} from '@elysiajs/cors'
+import { databasePlugin } from "./decorators/db"
 
 const app = new Elysia()
-.state('version',1)
-.decorate('getTodoById', (id: number)=> data.find(item => item.id === id))
-.decorate('addNewTodoItem', (title: string, description: string) => {
-  const newObject: { id: number, title: string, description: string, isComplete: boolean } = {
-    id: 0,
-    title,
-    description,
-    isComplete: false
-  }
-
-  const lastItem = data[data.length - 1]
-  const newId = lastItem ? lastItem.id + 1 : 1
+.use(cors())
+.use(databasePlugin)
 
-  newObject.id = newId
-
-  // Add the new object to the data array
-  data.push(newObject)
+app.state('version',1)
+.get("/all", ({ dbAction, database })=> {
+  const result = dbAction({
+    database: database(),
+    queryMethod: 'all',
+    query: "SELECT * FROM todos",
+    errorMessage: "Could not get all todo items",
+  })
 
-  // Return the modified data
-  return { data }
+  return result
 })
-.decorate('editTodoById', (id: number, updatedData: { title?: string; description?: string; isComplete?: boolean }) => {
-  // Find the index of the object with the specified ID
-  const todoIndex = data.findIndex(item => item.id === id);
 
-  if (todoIndex === -1) {
-    throw new Error("Could not find todo item")
-  }
-  // Perform the edit based on the provided properties in updatedData
-  data[todoIndex] = {
-    ...data[todoIndex],
-    ...updatedData
-  }
+.get('/todo/:id', ({params: {id}, dbAction, database})=>{
+  const result = dbAction({
+    database: database(),
+    queryMethod: 'get',
+    query: "SELECT * FROM todos WHERE id = ?",
+    params: [id],
+    errorMessage: "Could not find todo item",
+  })
 
-  return { data }
+  return result
 })
-.decorate('deleteTodoById', (id: number)=> {
-  data = data.filter(item => item.id !== id);
-  return data
 
-})
-.get("/all", ()=> {
-  return { data }
-})
-.get('/todo/:id', ({params: {id}, getTodoById})=>{
-  const response = getTodoById(Number(id))
-  if (!response) {
-    throw new Error("Could not find todo item")
-  }
-  return response
-})
-.post("/todo", ({ body, addNewTodoItem }) => {
-  const { title, description } = body
-  return addNewTodoItem(title, description)
+.post("/todo", ({ body, dbAction, database }) => {
+  const result = dbAction({
+    database: database(),
```

Comment:
```suggestion
```
The same database is used in every action, so this could be removed and put into the decorator I guess.